src/python/include/Alpha_complex_interface.h

```diff
@@ -30,67 +26,35 @@ namespace Gudhi {
 namespace alpha_complex {
 
 class Alpha_complex_interface {
- private:
-  using Exact_kernel = CGAL::Epeck_d<CGAL::Dynamic_dimension_tag>;
-  using Inexact_kernel = CGAL::Epick_d<CGAL::Dynamic_dimension_tag>;
-  using Point_exact_kernel = typename Exact_kernel::Point_d;
-  using Point_inexact_kernel = typename Inexact_kernel::Point_d;
-
-  template <typename CgalPointType>
-  std::vector<double> pt_cgal_to_cython(CgalPointType& point) {
-    std::vector<double> vd;
-    for (auto coord = point.cartesian_begin(); coord != point.cartesian_end(); coord++)
-      vd.push_back(CGAL::to_double(*coord));
-    return vd;
-  }
-
-  template <typename CgalPointType>
-  static CgalPointType pt_cython_to_cgal(std::vector<double> const& vec) {
-    return CgalPointType(vec.size(), vec.begin(), vec.end());
-  }
-
  public:
-  Alpha_complex_interface(const std::vector<std::vector<double>>& points, bool fast_version)
-      : fast_version_(fast_version) {
-    if (fast_version_) {
-      ac_inexact_ptr_ = std::make_unique<Alpha_complex<Inexact_kernel>>(
-          boost::adaptors::transform(points, pt_cython_to_cgal<Point_inexact_kernel>));
+  Alpha_complex_interface(const std::vector<std::vector<double>>& points, bool fast_version, bool exact_version) {
+    if (points[0].size() == 3) {
```

Comment:
I think we also have the problem that when the points are in 3D but they all lie on the same plane, then CGAL's Alpha_shape_3 doesn't work.

Reply:
Exactly, this is the most critical point that I don't know how to handle.

Reply:
One way is to build the Alpha_shape_3 and then ask for its dimension. If it isn't 3, then start over with the dD code. Another way is to build Point_d from all the points and use Affine_rank_d to determine if the whole lot is really 3D.